uqcsbot/api.py

```diff
@@ -230,7 +220,15 @@ def _initialise(self):
                 for chan in page['channels']:
                     self._add_channel(chan)
 
-            self.reload_ims()
+            # The ims cover the direct messages between the bot and users
+            for page in self._bot.api.im.list.paginate():
+                for im in page['ims']:
+                    if im['is_user_deleted']:
+                        continue
+
+                    im['name'] = im['id']
+                    im['is_private'] = True
+                    self._add_channel(im)
```

Comment:
I'm not sure if there is much benefit to pre-loading all the IMs. There are a lot of people in the slack. Why not just cache on the first im (which it looks like you have done with `_on_im_created`)?

Reply:
Not sure I understand. The im_created will only trigger for new users joining the chat as that im is automatically made. The initialise is just there to capture the ims of pre existing users. I'm probably just misunderstanding something.

Reply:
Ah, so the ims list will only return people who have an im, not every person in the slack? If so then I'm fine with it.

Reply:
It only returns all ims that the bot has access to

Reply:
But I guess technically that is everyone in the slack? From my testing everyone gets an im channel with the bot. Unless that's different for larger slacks.

Reply:
that's only slackbot, uqcsbot will be treated as any other user and require manual creation

Reply:
afaik

Reply:
Hmm okay, whenever I created a new user to test they automatically got  channel but they also got a channel with every other user so I guess that must just be low member count channels